src/table/sticky-columns/use-sticky-columns.ts

```diff
@@ -171,56 +171,57 @@ export function useStickyCellStyles({
 }: UseStickyCellStylesProps): StickyCellStyles {
   const cellRef = useRef<HTMLElement>(null) as React.MutableRefObject<HTMLElement>;
   const setCell = stickyColumns.refs.cell;
+
+  // unsubscribeRef to hold the function to unsubscribe from the store's updates
+  const unsubscribeRef = useRef<null | (() => void)>(null);
+
+  // refCallback updates the cell ref and sets up the store subscription
   const refCallback = useCallback(
     node => {
+      if (unsubscribeRef.current) {
+        // Unsubscribe before we do any updates to avoid leaving any subscriptions hanging
+        unsubscribeRef.current();
+      }
+
+      // Update cellRef and the store's state to point to the new DOM node
       cellRef.current = node;
       setCell(columnId, node);
+
+      // If the node is not null (i.e., the table cell is being mounted or updated, not unmounted),
+      // set up a new subscription to the store's updates
+      if (node) {
+        unsubscribeRef.current = stickyColumns.store.subscribe(selector, (newState, prevState) => {
+          updateCellStyles(selector(newState), selector(prevState));
+        });
+      }
     },
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I think it will be safer to declare the `updateCellStyles` inside the ref callback and add `store` to the dependency array. Otherwise if there would be a new dependency for updateCellStyles it might result in a bug which would be very difficult to track.